File: frontend/lib/market-learning-roadmap.ts
export type MarketLearningRoadmapTask = {
  task_key: string;
  task_type: "resolve_evidence" | "submit_evidence" | "explore_scenario" | "compare_risk_tier";
  priority: number;
  title: string;
  rationale: string;
  completion_criteria: string;
  reflection_prompt: string;
  scenario_name: string | null;
  risk_tier: string | null;
  session_id: number | null;
  evidence_id: number | null;
};

export type MarketLearningRoadmap = {
  status: "not_started" | "active" | "complete";
  calculation_criteria: string[];
  tasks: MarketLearningRoadmapTask[];
  disclaimer: string;
};

File: backend/app/schemas/market_learning_roadmap.py
from typing import Literal

from pydantic import BaseModel


class MarketLearningRoadmapTask(BaseModel):
    task_key: str
    task_type: Literal["resolve_evidence", "submit_evidence", "explore_scenario", "compare_risk_tier"]
    priority: int
    title: str
    rationale: str
    completion_criteria: str
    reflection_prompt: str
    scenario_name: str | None = None
    risk_tier: str | None = None
    session_id: int | None = None
    evidence_id: int | None = None


class MarketLearningRoadmapRead(BaseModel):
    status: Literal["not_started", "active", "complete"]
    calculation_criteria: list[str]
    tasks: list[MarketLearningRoadmapTask]
    disclaimer: str
